### ISSUE TITLE
Confusing axiom on FOODON `chemical food component`

### ISSUE THREAD

**Issue by kaiiam** [FoodOntology/foodon#191]
I noticed that `chemical food component` http://purl.obolibrary.org/obo/FOODON_03411041 has the following axioms:

<img width="435" alt="image" src="https://user-images.githubusercontent.com/12255688/153367442-220f1690-8f46-4844-ab2a-7d4872d0d5d3.png">

I suspect a fishy pattern although perhaps it is correct. In natural language I can see that these are themselves food materials that can also be part of other food food materials. Perhaps it is correct but I wonder if by your definition of `food material` `chemical food component` should actually be a subclass:

`food material`

> Any substance that can be consumed by an organism to satisfy nutritional or other health needs, or to provide a social or organoleptic food experience

`chemical food component`

> Any chemical or chemical mixture that exists in a food material or was added to a food material.

I wonder if there might be a better hierarchical structure for this breaking up the above into chemical components that might be in food materials (e.g the new CDNO `dietary nutritional component` hierarchy vs those chemicals intentionally added to food (preservatives etc). Not sure if that's helpful but I think there might be future use cases for searching specifically for or differentiating the added chemicals from the chemicals part of the food.

Maybe I'm off here but perhaps something to consider. 

Another quick note some FOODON defs seem to be duplicated at the moment at least in OLS not in ontobee though. 

<img width="500" alt="image" src="https://user-images.githubusercontent.com/12255688/153370284-40bce29b-8daf-468a-b3ab-f65dc8d95bfa.png">

**Comment by ddooley** [comment]
Thanks for feedback! First, it looks like duplication issue has gone away, not sure how it arose.  

Point taken that it sounds odd to be a subclass of, and part of at same time, and I'm thinking since 'part of' includes self too (i.e. not a proper part), that its a bit redundant and should be taken out. It isn't really doing anything.

At the level of chemical ingredients (i.e. ingredients that are pretty much a single chemical, maybe with some impurity), there are different categories that arise.  In food industry a "food additive" is a type of ingredient that achieves some functional objective in a food, and this actually includes fortificants sometimes aimed to boost levels of nutrients that are naturally occurring or are absent.  If I'm not mistaken additives still don't cover all intentionally added chemicals, such as salt, or water, which are "every day" type of ingredients.  

That aside, there are chemical constituents as you say that are not added, but occur naturally in food.  I thought the "chemical food component" subclass structure would handle a number of these distinctions, but its an ongoing curation project to finesse these and get all chemicals into at least a primary superclass here. I suppose we could make a "naturally occurring chemical food component" class to distinguish that case, but it would perhaps be only for typing instance chemicals with respect to some instance of food material.

**Comment by kaiiam** [comment]
@LilyAndres discussed a similar issue today in CDNO regarding the distinction of whether or not something is nutritional with regard to being dietary. Currenty CDNO's [dietary nutritional component](https://www.ebi.ac.uk/ols/ontologies/cdno/terms?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FCDNO_0000001) is doing both but includes subclasses like lead or gold which get ingested but don't have a nutritional value. We're thining about how to handle this distinction in CDNO which could help with the problem of chemical constituents in food. 

One possibility would be to modify our CDNO hierarchy to be:

```
material entity
    dietary  component (is eaten)
        dietary nutritional component (is eaten and provides nutrition)
``` 

Or we could add a role quality or disposition called `nutritional` and add relations to that role to materials we want to say provide nutrients:

```
material entity
    dietary  component
        carbohydrate (add axiom has role some nutritional role)
role
    nutritional role
```

Sorry to hijack the thread but I think these problems might converge here. Let me know what you think @ddooley.

**Comment by maweber-bia** [comment]
Hi, I think it is useful to clarify this point in order to be consistent between FoodON and CDNO; besides, the concept "Chemical food component (http://purl.obolibrary.org/obo/FOODON_03411041) is also integrated in ENVO.

to my mind, it is simplest to distinguish role and entity so the second option seems good to me (it allows to add axiome has some nutritional role once the evidence is provided that the component actually has a nutritional role)



> At the level of chemical ingredients (i.e. ingredients that are pretty much a single chemical, maybe with some impurity), there are different categories that arise. In food industry a "food additive" is a type of ingredient that achieves some functional objective in a food, and this actually includes fortificants sometimes aimed to boost levels of nutrients that are naturally occurring or are absent. If I'm not mistaken additives still don't cover all intentionally added chemicals, such as salt, or water, which are "every day" type of ingredients.

Yes, In food industry a "food additive" is a type of ingredient that achieves some functional objective in a food --> this could also be treated as a role (additive role or functional role)

@Damion, you are right, additives still don't cover all intentionally added chemicals, such as salt, or water, which are "every day" type of ingredients.  Besides, you can also have "ingredients" which are composed of ingredients and there is a category of chemicals that are classified as "processing aids" but not considered as ingredients nor "additives" according to the European regulation.


cheers,
Magalie

**Comment by maweber-bia** [comment]
> I suppose we could make a "naturally occurring chemical food component" class to distinguish that case, but it would perhaps be only for typing instance chemicals with respect to some instance of food material.

this is perhaps useful, since you also have "neoformed compounds" that result from a process

**Comment by ddooley** [comment]
@kaiiam I see both your solutions are useful, but now am understanding that the latter role-focused approach is probably a better primary home for things, with roles not always being exercised.  The former approach's eaten + nutrition classes then may be useful as they pertain to the typing of instance data, rather than having classes defined under them.  I.e. "I ate this bowl of porridge containing milk, and that was nutritious for me." (so I'm not lactose intolerant!)

**Comment by LilyAndres** [comment]
Thanks for posting the issue @kaiiam. I'm thinking that maybe the second option would be better, as we now have the 'dietary role' class in CDNO and we could add 'nutrition role' in it? I will discuss this with the team on Monday.

**Comment by kaiiam** [comment]
@maweber-bia 

> additives still don't cover all intentionally added chemicals, such as salt, or water, which are "every day" type of ingredients...

Yes we should try to cover these distinctions in CDNO and or FOODON. 

@ddooley 

>  The former approach's eaten + nutrition classes then may be useful as they pertain to the typing of instance data, rather than having classes ... lactose intolerant 

Yes that's a perfect rational for not trying to force everything into a "nutritional component" hierarchy. Would using the 2nd option with roles also have that problem? If assert an axiom like `lactose has role some nutritional role` aren't we also saying that lactose always provides nutrition? Perhaps it's not. The definition of [BFO:role](http://purl.obolibrary.org/obo/BFO_0000023) is:

> A realizable entity the manifestation of which brings about some result or end that is not essential to a continuant in virtue of the kind of thing that it is **but that can be served or participated in** by that kind of continuant in **some kinds** of natural, social or institutional **contexts**.

It seems there are enough qualifiers in role's definition to say that it's not always the case? I'm not sure though. Thoughts anyone? I wonder if BFO:[disposition](http://purl.obolibrary.org/obo/BFO_0000016) or [function](http://purl.obolibrary.org/obo/BFO_0000034) are more appropriate?

**Comment by kaiiam** [comment]
@ramonawalls would you mind weighing in about the placement of "nutritional" what would be the most appropriate placement within BFO?

**Comment by ramonawalls** [comment]
My two cents:

```
b is a disposition means: b is a realizable entity & b’s bearer is some material entity & b is such that if it ceases to exist, then its bearer is physically changed, & b’s realization occurs when and because this bearer is in some special physical circumstances, & this realization occurs in virtue of the bearer’s physical make-up. (axiom label in BFO2 Reference: [062-002])

b is a role means: b is a realizable entity & b exists because there is some single bearer that is in some special physical, social, or institutional set of circumstances in which this bearer does not have to be& b is not such that, if it ceases to exist, then the physical make-up of the bearer is thereby changed. (axiom label in BFO2 Reference: [061-001])
```
The key difference between roles and dispositions is that if the role ceases to exist, the bearer does not change physically, whereas if the disposition ceases to exist, the bearer is changed physically. 

To use the classic example, think of an urn-shaped object that has the disposition to be fragile and is holding flowers and therefore realizing the role of a vase. If I coat the urn in rubber, it is no longer fragile. The disposition is lost and the bearer is physically changed. If I take the flowers out of the urn, it is no longer realizing the vase role. The role is gone, but the bearer is not physically changed.

Now look at the example of lactose as a nutrient. Lactose has the potential to be nutritious. That potential is realized when it is consumed by someone who tolerates lactose. If you take away lactose's ability to provide nutrition when eaten by a lactose tolerant person, the realizable entity will be gone. I think the only way that you could change the potential to provide nutrition to lactose tolerant people is by changing the physical structure of lactose. Therefore, the bearer has changed, and that makes the realizable entity a disposition.

In order to say that nutrition is a function, you would have to demonstrate that the bearer specifically evolved to provide that realizable entity (e.g., that lactose evolved to be nutritious). While it may be true, I would not want the burden of having to prove it for every thing I wanted to give the nutrition role to, and there are definitely nutritious chemicals that did not evolve to provide nutrition (e.g., most carbohydrates in plants).

**Comment by kaiiam** [comment]
Great explanation thank you @ramonawalls!

Based on that I propose CDNO should 

1) have a `nutritional disposition` class, with a definition like:

> A role which 1) inheres in a material entity 2) is realized when that entity is taken in by an organism and contributes to the survival, growth, development, or other biological function of such organism, its bionts, or its holobionts.

2) change `dietary nutritional component` to `dietary component` and modify it's definition to be something like:

> A material entity which is taken in by an organism during a dietary process. 

3) Add the axiom `has disposition some nutritional disposition` to the relevant classes in the CDNO dietary component hierarchy making sure not to assign the axiom to terms like `lead` or `gold` etc. 

Similarly, back to the original discussion of `chemical food component` perhaps FOODON could differentiate food additives from processing aid or other legislative differentiation via role classes. e.g. `citric acid has role some food additive role`. I think that would allow the flexibility of saying citric acid can serve the role of food additive but it isn't always (it occurs naturally in oranges). I think that might be better than the current FOODON assertion that citric acid is a food additive (via the subclass relation), which logically I think means citric acid is always a food additive which is not the case. 

Thoughts?

**Comment by maweber-bia** [comment]
> I think the only way that you could change the potential to provide nutrition to lactose tolerant people is by changing the physical structure of lactose.

 @ramonawalls
I think the correct sentence is

I think the only way that you could change the potential to provide nutrition to lactose **in**tolerant people is by changing the physical structure of lactose 

is that ok ?

**Comment by kaiiam** [comment]
@maweber-bia I think ramona meant to say lactose tolerant people. She wasn't talking about lactose intolerance more about lactose being able to provide nutrition (to people who can tolerate it).

**Comment by maweber-bia** [comment]
> Similarly, back to the original discussion of `chemical food component` perhaps FOODON could differentiate food additives from processing aid or other legislative differentiation via role classes. e.g. `citric acid has role some food additive role`. I think that would allow the flexibility of saying citric acid can serve the role of food additive but it isn't always (it occurs naturally in oranges). I think that might be better than the current FOODON assertion that citric acid is a food additive (via the subclass relation), which logically I think means citric acid is always a food additive which is not the case.

Yes, it seems good but just to be sure the "role" you mention here for citric acid would be a bfo:role or a bfo:disposition ?
A role is external to the bearer, whereas a disposition is internally-grounded

I have read in a conference by B. SMith about  Cause-effect relationship that 
"Causes an effect means that there is some independent continuant which has a certain disposition which is triggered by a certain process and that triggering is the cause of the process which gets triggered which is the effect"




--> the citric acid is used as an additive to acidify the food and so it has  an acidification role which inheres in the form of the molecule/substance added ? or do we need to consider that some chemical reaction will occur that will change the physical form of the molecule citric acid ?

--> processing aids are defined as substances which are intentionnally added to the food during the production/transformation process but that should not be present (or detectable) in the final product (and that is the reason why they are not listed as "additives" so far in the European legislation, but this is perhaps a kind of "exceptional rule")

--> ingredients are the substances which are mixed together to make food. They can be either from natural sources or "additives", and ingredients should be listed in the nutrition facts as such

![image](https://user-images.githubusercontent.com/64576528/155687250-154d5c1b-a44d-45a6-acf5-2d3ebc94f3ca.png)

**Comment by kaiiam** [comment]
> Yes, it seems good but just to be sure the "role" you mention here for citric acid would be a bfo:role not a disposition ?

Yes I think role is more appropriate for the food additive case and disposition for the nutritional case.  

> If I take the flowers out of the urn, it is no longer realizing the vase role. The role is gone, but the bearer is not physically changed.

similarly if we stop adding citric acid to food it's food additive role is gone but the bearer (the citric acid) is not physically changed. 

> [A] urn-shaped object that has the disposition to be fragile ... if I coat the urn in rubber, it is no longer fragile. 

lactose has the disposition to be nutritious (to lactose tolerant people) if we were to change lactose itself at a chemical level it might no longer be nutritious.  

nutritional disorders e.g. [lactose intolerance](http://purl.obolibrary.org/obo/MONDO_0100345) are covered in other OBO ontologies and are not in scope for FOODON or CDNO.

**Comment by maweber-bia** [comment]
> lactose has the disposition to be nutritious (to lactose tolerant people) if we were to change lactose itself at a chemical level it might no longer be nutritious.
> 
> nutritional disorders e.g. [lactose intolerance](http://purl.obolibrary.org/obo/MONDO_0100345) are covered in other OBO ontologies and are not in scope for FOODON or CDNO.

yes, we should only consider "healthy people" to define the dispostion to be nutritious in FoodON
(here lactose tolerant people only)

**Comment by kaiiam** [comment]
Yes it's more about what lactose can do rather than how everyone reacts to it.

**Comment by ddooley** [comment]
I appreciate the role vs disposition discussion.  Agreed that a chemical can have a role like an additive role as an input to a process - a role that can pertain some of the time but not necessarily all of the time.  (Example?: citric acid is an acidifier but not if its being added to lemon juice which is already quite acidic).  But note these roles are relative to chemicals and the food they are in. There appears to be a plug-and-play aspect to these roles. i.e. a number of chemicals can fulfill the role.

When it comes to "nutrient", we are talking about a chemical with respect to its consumer (absorber), and the food is more of a medium (most of the time; Counterexample?)  A nutrient is not so plug-and-play, without it, the consumer suffers.  So that's why we're drawn to the idea that a chemical may bear a "nutrient disposition" with respect to a consumer.  Without the chemical, the consumer is changed, for the worse usually unless dealing with a rare disease.

As an aside, about food ingredients, do we have cases where a food item's identity depends on certain ingredients having a disposition such that the food without them would be essentially changed? E.g.  FoodOn's "has defining ingredient" singles out certain ingredients as having a special role, for example one can't have an omelette without eggs.  So I guess that would be an "essential ingredient disposition" :)

**Comment by kaiiam** [comment]
> one can't have an omelette without eggs. So I guess that would be an "essential ingredient disposition"

I think in that case if we want to state for a fact that an omelet requires eggs, then perhaps just a `has part some egg` might suffice?

**Comment by maweber-bia** [comment]
I think that it is difficult to distinguish between defining ingredients and "non-defining" ingredients (let's just say "ingredient") 

in the case of eggs and omelette it can be obvious but what in the case of "cheese" or "yogurt" or "wine" ? For these products, they may exist different regulations establishing a list of "mandatory" or "forbidden" ingredients and the regulations can differ from one region to another... 

I think we should think about the notion of composite food as a mixture of ingredients and/or additives but that it is difficult to establish a property "has defining ingredient" which will be always applicable at a generic level, and defining  a role would be more suitable  ?

**Comment by ddooley** [comment]
What arose for me in above discussion was relationship between those defining ingredients and whether they had therefore a role or a disposition inhering in them with respect to a food product. Indeed @maweber-bia a 'has defining ingredient' relation is a shortcut for some ingredient having an "essential" role or disposition with respect to a food product.  You raise an interesting point - where we have legislation mandating ingredients or absence of them, that probably indicates socially proscribed roles are at work rather than "inbuilt" dispositions?!  But then, if disposition has an identity dependency aspect to it, essentially saying identity of an entity depends on processes which have certain inputs, e.g. nutrients to life support of an organism, then perhaps we can apply disposition not only in a biological sense, but as well to a more abstract informatic or behaviour space where product appellations have a dependency on legislation/regulation.  A legislative disposition. Hmm - but where regulation is actually defining an entity, maybe reference to disposition isn't even needed?

I had introduced to FoodOn the 'has ingredient' and more specialized 'has defining ingredient' object properties as a way of attaching food products to their ingredients, and giving some prominence - that everyone would easily agree on - to some of those ingredients.  I thought it would be one navigational way to approach food products, but probably not much value beyond that.  So potato leek soup has defining ingredients potato and leek; and one could explore all the other things potato or leek is essential in.  I recognize there are plenty of areas that we'd probably not venture 'has defining ingredient' for.  That was going to be as far as FoodOn could go in having any generalized (true for everyone) multi-component food product composition.

As an aside: Now there's a little complication about the difference between ingredient and 'has part'.  Ingredient is something one adds to a mixture, but whether a mixture has part some substance further on down the processing chain is another question.  Baking soda for example, transforms, so although it is an ingredient, it isn't a part of muffins.  But some ingredients maintain their form and can be considered part of some food end product - peanuts in peanut brittle perhaps. Most other ingredients undergo some identity transformation (chopping, cooking, etc.) that would make an OWL reasoner protest that what goes into a food transformation process (parts and all) and what comes out are not the same thing (equivalent), with transformations often occurring in the parts, not just the whole.

**Comment by maweber-bia** [comment]
> I thought it would be one navigational way to approach food products, but probably not much value beyond that.

That was a user-friendly approach but we should go further, if we can make clear what an "ingredient" is in an ontological way



> As an aside: Now there's a little complication about the difference between ingredient and 'has part'. Ingredient is something one adds to a mixture, but whether a mixture has part some substance further on down the processing chain is another question. Baking soda for example, transforms, so although it is an ingredient, it isn't a part of muffins. But some ingredients maintain their form and can be considered part of some food end product - peanuts in peanut brittle perhaps. Most other ingredients undergo some identity transformation (chopping, cooking, etc.) that would make an OWL reasoner protest that what goes into a food transformation process (parts and all) and what comes out are not the same thing (equivalent), with transformations often occurring in the parts, not just the whole.

Baking soda, for example, transforms, so although it is an ingredient, it is not part of the muffins 
--> that's right, that's why it should be associated with a "functional role" and not as an ingredient (just like processing aids).

Most other ingredients undergo some identity transformation (chopping, cooking, etc.) that would make an OWL reasoner protest that what goes into a food transformation process (parts and all) and what comes out are not the same thing (equivalent), with transformations often occurring in the parts, not just the whole.
--> I do not think that we have to define all the input as "ingredients" : we have input components (which are entities, namely food material) and output components which are **other entities** (other food materials)

Is that possible for an OWL reasoner ?



> But then, if disposition has an identity dependency aspect to it, essentially saying identity of an entity depends on processes which have certain inputs, e.g. nutrients to life support of an organism, then perhaps we can apply disposition not only in a biological sense, but as well to a more abstract informatic or behaviour space where product appellations have a dependency on legislation/regulation. A legislative disposition. Hmm - but where regulation is actually defining an entity, maybe reference to disposition isn't even needed?

I still don't know if the disposition or the role is the best solution! But I have still in mind that "Causes an effect means that there is some independent continuant which has a certain disposition which is triggered by a certain process and that triggering is the cause of the process which gets triggered which is the effect" and this has to do with dispositions (according to Barry Smith)

If you are adding Baking soda, it causes an effect, doesn't it ? and you are adding this intentionally, for a specific objective (on the instance level)

Thoughts ?

**Comment by ddooley** [comment]
So perhaps we need to formalize "ingredient" a bit more for food products, or define a few term variants depending on technical or home-cooking contexts.  So this reminds me @maweber-bia that you were zeroing in on this in #57 so I'll take discussion over there on defining "ingredient".

**Comment by LilyAndres** [comment]
Hey, for what I can see in issue #57, it's a bit more clear what could be the next step for FoodOn. 

From @ddooley 
>The former approach's eaten + nutrition classes then may be useful as they pertain to the typing of instance data, rather than having classes ... lactose intolerant

Yes, so FoodOn can adopt the same idea from CDNO to add axioms (nutritional role or has role some food additive).


 @kaiiam thanks for the proposals, I think this will be very helpful.
> 1. have a nutritional disposition class, with a definition like:
A role which 1) inheres in a material entity 2) is realized when that entity is taken in by an organism and contributes to the survival, growth, development, or other biological function of such organism, its bionts, or its holobionts.
> 2. Change **dietary nutritional component** to **dietary component** and modify it's definition to be something like:
A material entity which is taken in by an organism during a dietary process.

We still have some pending conversations around the fact that some components can be nutritious but they are not ingested in the diet, but via "Enteral" and "Parenteral". It might not be the scope of CDNO, just important to remember.

> 3. Add the axiom has disposition some nutritional disposition to the relevant classes in the CDNO dietary component hierarchy making sure not to assign the axiom to terms like lead or gold etc.

I will try to start to work on this asap.
